cats/cats-redis/src/main/java/com/netflix/spinnaker/cats/redis/cache/RedisCache.java

```diff
@@ -72,28 +91,25 @@ public void merge(String prefix, String type, int itemCount, int keysWritten, in
     private final String prefix;
     private final JedisSource source;
     private final ObjectMapper objectMapper;
-    private final int maxMsetSize;
-    private final int maxMergeCount;
     private final CacheMetrics cacheMetrics;
-    private final boolean enableHashing;
-
-    public RedisCache(String prefix, JedisSource source, ObjectMapper objectMapper, int maxMsetSize, int maxMergeCount, boolean enableHashing, CacheMetrics cacheMetrics) {
-        Preconditions.checkArgument(
-          maxMsetSize % 2 == 0, String.format("maxMsetSize must be even (%s)", maxMsetSize)
-        );
+    private final RedisCacheOptions options;
 
+    public RedisCache(String prefix, JedisSource source, ObjectMapper objectMapper, RedisCacheOptions options, CacheMetrics cacheMetrics) {
         this.prefix = prefix;
         this.source = source;
         this.objectMapper = objectMapper.disable(SerializationFeature.WRITE_NULL_MAP_VALUES);
-        this.maxMsetSize = maxMsetSize;
-        this.maxMergeCount = maxMergeCount;
-        this.enableHashing = enableHashing;
+        this.options = options;
         this.cacheMetrics = cacheMetrics == null ? new CacheMetrics.NOOP() : cacheMetrics;
     }
 
     @Override
+    public void merge(String type, CacheData item) {
```

Comment:
there's a bit of reformatting noise in this changeset - I reordered the methods to reflect the call chain:
merge calls mergeAll calls mergeItems
evict calls evictAll calls evictItems
get calls getAll calls getItems